### PR TITLE
add subscription cancellation for google platform

### PR DESCRIPTION
**Pull request by adrian-gierakowski**
this is a rebased version of #30 with conflicts resolved

I couldn't rebase the branch used for #30 since I currently use a SHA from that branch in package.json in production

### CODE REVIEW

File: lib/google/index.js [hunk 1]
```diff
@@ -87,3 +96,43 @@ exports.verifyPayment = function (payment, cb) {
...
+
+	try {
+		keyObject = validatePaymentAndParseKeyObject(payment);
+	} catch (error) {
+		return process.nextTick(function () {
+			cb(error);
+		});
+	}
+
+	/* jshint camelcase:false */
```
Comment (ronkorving) [inline lib/google/index.js]: We use ESLint now, so this probably doesn't do much

File: index.js [hunk 2]
```diff
@@ -35,3 +35,35 @@ exports.verifyPayment = function (platform, payment, cb) {
 		cb(null, result);
 	});
 };
+
+
+exports.cancelSubscription = function (platform, payment, cb) {
+	function syncError(error) {
```
Comment (ronkorving) [inline index.js]: I guess you meant `asyncError`?

File: index.js [hunk 3]
```diff
@@ -35,3 +35,35 @@ exports.verifyPayment = function (platform, payment, cb) {
...
+		process.nextTick(function () {
+			cb(error);
+		});
+	}
+
+	if (!payment) {
+		return syncError(new Error('No payment given'));
```
Comment (ronkorving) [inline index.js]: This is absolutely not a requirement, but did you know you could write:

```js
return process.nextTick(cb, new Error('No payment given'));
```

It's not much longer, and you don't really need that `syncError` helper function. Just a suggestion, not a blocker.

**Comment by ronkorving** [comment]
I'm very happy with this. Left a few comments for your consideration though.

File: lib/google/index.js (hunk 1 above)
Comment (adrian-gierakowski) [reply lib/google/index.js]: removed

File: index.js (hunk 2 above)
Comment (adrian-gierakowski) [reply index.js]: I created `cancelSubscription` by simply copying and adapting `verifyPayment`. I could improve both but maybe in a seperate PR?

File: index.js (hunk 3 above)
Comment (adrian-gierakowski) [reply index.js]: a above (https://github.com/Wizcorp/node-iap/pull/55/files#r218949991)

Comment (ronkorving) [reply index.js]: Got it, in that case let's land this first :)
Follow-up PR always appreciated of course! :)

**Comment by ronkorving** [comment]
Thanks!

**Pull request merged**